dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,387 @@
+
+            // Setting NumberFormatter to something that isn't an INumberParser will throw an exception, so this should be safe
+            auto numberParser = NumberFormatter().as<winrt::INumberParser>();
+            auto parsedNum = numberParser.ParseDouble(text);
+
+            if (!parsedNum)
+            {
+                if (BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten)
+                {
+                    // Override text to current value
+                    UpdateTextToValue();
+                }
+            }
+            else
+            {
+                Value(parsedNum.Value());
+            }
+        }
+    }
+}
+
+void NumberBox::OnSpinDownClick(winrt::IInspectable const&  sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(false);
+}
+
+void NumberBox::OnSpinUpClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(true);
+}
+
+void NumberBox::OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args)
+{
+    switch (args.OriginalKey())
+    {
+        case winrt::VirtualKey::Enter:
+        case winrt::VirtualKey::GamepadA:
+            ValidateInput();
+            break;
+
+        case winrt::VirtualKey::Escape:
+        case winrt::VirtualKey::GamepadB:
+            UpdateTextToValue();
+            break;
+
+        case winrt::VirtualKey::Up:
+            StepValue(true);
+            break;
+
+        case winrt::VirtualKey::Down:
+            StepValue(false);
+            break;
+    }
+}
+
+void NumberBox::OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args)
+{
+    if (HyperScrollEnabled())
+    {
+        int delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
+        if (delta > 0)
+        {
+            StepValue(true);
+        }
+        else if (delta < 0)
+        {
+            StepValue(false);
+        }
+    }
+}
+
+void NumberBox::StepValue(bool isPositive)
+{
+    double oldVal = Value();
+    double newVal = oldVal;
+
+    if (isPositive)
+    {
+        newVal += StepFrequency();
+    }
+    else
+    {
+        newVal -= StepFrequency();
+    }
+
+    if (WrapEnabled())
+    {
+        if (newVal > Maximum())
+        {
+            newVal = Minimum();
+        }
+        else if (newVal < Minimum())
+        {
+            newVal = Maximum();
+        }
+    }
+
+    // Safeguard for floating point imprecision errors
+    int StepFreqSigDigits = ComputePrecisionRounderSigDigits(newVal);
+    m_stepPrecisionRounder.SignificantDigits(StepFreqSigDigits);
+    newVal = m_stepPrecisionRounder.RoundDouble(newVal);
+
+    // Update Text and Revalidate new value
+    Value(newVal);
+}
+
+// Check if text resembles formulaic input to determine if parser should be executed
+bool NumberBox::IsFormulaic(const winrt::hstring& in)
+{
+    std::wstring input(in);
+    std::wregex formula(L"^([0-9()\\s]*[+-/*^%]+[0-9()\\s]*)+$");
```

Comment:
>L"^([0-9()\\s]*[+-/*^%]+[0-9()\\s]*)+$" [](start = 24, length = 39)

These regex's should be raised into static constexpr on the class rather than locals that get created each time